cpp/ql/src/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -156,8 +156,65 @@ float safeFloor(float v) {
   result = v
 }
 
+/** A `MulExpr` where exactly one operand is constant. */
+private class MulByConstantExpr extends MulExpr {
+  float constant;
+  Expr operand;
+
+  MulByConstantExpr() {
+    exists(Expr constantExpr |
+      this.hasOperands(constantExpr, operand) and
+      constant = constantExpr.getFullyConverted().getValue().toFloat() and
```

Comment:
It might be better to use the `getValue` predicate defined further up in the file as it handles slightly more cases (see https://github.com/github/codeql/pull/3026).

Reply:
I've pushed a commit to change all calls to use that predicate, including calls that already existed (used in the `>>` support).